web-client/src/views/FileDocument/StateDrivenFileInput.tsx

```diff
@@ -48,6 +61,61 @@ export const StateDrivenFileInput = connect<
 
     const fileOnForm = file || form[fileInputName] || form.existingFileName;
 
+    // Setting the filename here so that we can display it before validation
+    // finishes, otherwise a slow machine might have slight lag and allow the user
+    // to "choose file" again.
```

Comment:
Maybe this is overkill?

Reply:
Here is an example of the issue I can get when throttling my browser _without_ `selectedFilename`. (Slight delay between valid selection and updated UI leads to "Choose file" being able to be selected again.)


https://github.com/user-attachments/assets/a5574ccc-160e-41aa-83ad-21c6a6daa205